schema.go

```diff
@@ -892,7 +904,7 @@ func NewEnumSchema(name, namespace string, symbols []string, opts ...SchemaOptio
 		return nil, errors.New("avro: enum must have a non-empty array of symbols")
 	}
 	for _, sym := range symbols {
-		if err = validateName(sym); err != nil {
+		if err = validateName(sym, cfg.skipNameValidation); err != nil {
```

Comment:
Would it make sense to check `skipNameValidation` at this point once rather than in `validateName` for every symbol? Something like:
```go
if !cfg.skipNameValidation {
	for _, sym := range symbols {
		if err = validateName(sym); err != nil {
			return nil, fmt.Errorf("avro: invalid symbol %q", sym)
		}
	}
}
```
I don't like the extra nesting tbh and I don't have strong feelings about it anyway. Maybe it better matches the "context" in `validateName`.